### PR TITLE
New Subworkflow: Bam_trim_primers_ivar

### PR DESCRIPTION
**Pull request by Joon-Klaps**
## PR checklist

Addresses first bullet from https://github.com/nf-core/viralrecon/issues/371  

adds the subworkflow bam_trim_primers_ivar used in nf-core/viralrecon to nf-core/modules

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: subworkflows/nf-core/bam_trim_primers_ivar/main.nf [hunk 1]
```diff
@@ -0,0 +1,44 @@
+include { IVAR_TRIM               } from '../../../modules/nf-core/ivar/trim/main'
+include { BAM_SORT_STATS_SAMTOOLS } from '../bam_sort_stats_samtools/main'
+
+workflow BAM_TRIM_PRIMERS_IVAR {
+    take:
+    bam   // channel: [ val(meta), [ bam ], [bai] ]
+    bed   // path   : bed
+    fasta // channel: reference.fasta
```
Comment (nvnieuwk) [inline subworkflows/nf-core/bam_trim_primers_ivar/main.nf:8]:
```suggestion
    ch_bam   // channel: [ val(meta), path(bam), path(bai) ]
    ch_bed   // channel: [ path(bed) ]
    ch_fasta // channel: [ path(fasta) ]
```

Can you also change the output comments to this format? All channels should start with `ch_` (except for the output names of the subwf) and all values should start with `val_`

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: subworkflows/nf-core/bam_trim_primers_ivar/meta.yml
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply subworkflows/nf-core/bam_trim_primers_ivar/meta.yml:1]: Aah... I forgot about this one

[the remark this answers is not in the document]
Comment (nvnieuwk) [reply subworkflows/nf-core/bam_trim_primers_ivar/meta.yml:1]: No problem!

File: subworkflows/nf-core/bam_trim_primers_ivar/main.nf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    //
+    BAM_SORT_STATS_SAMTOOLS (
+        IVAR_TRIM.out.bam,
+        ch_fasta
+    )
+
+    ch_versions = ch_versions.mix(BAM_SORT_STATS_SAMTOOLS.out.versions.first())
+
+    emit:
+    bam_orig = IVAR_TRIM.out.bam                    // channel: [ val(meta), bam   ]
```
Comment (nvnieuwk) [inline subworkflows/nf-core/bam_trim_primers_ivar/main.nf:33]:
```suggestion
    bam_orig = IVAR_TRIM.out.bam                    // channel: [ val(meta), path(bam) ]
```

Can you add paths to all the files in the comments (see above for an example :))

File: subworkflows/nf-core/bam_trim_primers_ivar/meta.yml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      documentation: https://andersen-lab.github.io/ivar/html/manualpage.html
+      licence: ["GPL-3.0-or-later"]
+
+input:
+  - ch_bam:
+      type: file
+      description: |
+        Structure: [ val(meta), path(bam), path(bai) ]
+        Groovy Map containing sample information
+        e.g. [ id:'test' ], BAM/CRAM/SAM file, index file for co-ordinate sorted BAM file
```
Comment (nvnieuwk) [inline subworkflows/nf-core/bam_trim_primers_ivar/meta.yml:35]: This is awesome! Thanks 🥳

**Review by nvnieuwk — COMMENTED** [review]
One more small comment and then you are good to go!

**Comment by Joon-Klaps** [comment]
Not Sure why conda test is failing though..

**Comment by nvnieuwk** [comment]
It seems to be something tool specific. Maybe a wrong conda package?

**Comment by Joon-Klaps** [comment]
> It seems to be something tool specific. Maybe a wrong conda package?

Yeah... doesn't seem something I'll be able to finish as part of this PR... https://github.com/andersen-lab/ivar/issues/93

**Pull request closed without merging**